src/main/java/pro/sky/JavaCoredemo2/services/MathQuestionService.java

```diff
@@ -0,0 +1,58 @@
+package pro.sky.JavaCoredemo2.services;
+
+import pro.sky.JavaCoredemo2.entity.Question;
+import pro.sky.JavaCoredemo2.exceptions.NoQuestionsException;
+import pro.sky.JavaCoredemo2.repository.MathQuestionsRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.Collection;
+import java.util.Random;
+
+@Service
+public class MathQuestionService implements QuestionService {
+    private final Random random = new Random();
+    private final MathQuestionsRepository repository;
```

Comment:
```suggestion
    private final QuestionsRepository repository;
```
инъекцию надо всегда делать по интерфейсу, а не реализации

Reply:
также см. внимательнее задание про Qualifier аннотацию